substrate/frame/staking/src/tests.rs

```diff
@@ -4133,6 +4133,48 @@ fn test_multi_page_payout_stakers_by_page() {
 	});
 }
 
+#[test]
+fn unbond_with_chill_works() {
+	// Should test:
+	// * Given an account being bonded [and chosen as a validator](not mandatory)
+	// * it can force unbond a portion of its funds from the stash account.
+	ExtBuilder::default().nominate(false).build_and_execute(|| {
+		// Set payee to stash.
+		assert_ok!(Staking::set_payee(RuntimeOrigin::signed(11), RewardDestination::Stash));
+
+		// Give account 11 some large free balance greater than total
+		let _ = Balances::make_free_balance_be(&11, 1000000);
+
+		// confirm that 10 is a normal validator and gets paid at the end of the era.
+		mock::start_active_era(1);
+
+		// Initial state of 11
+		assert_eq!(
+			Staking::ledger(11.into()).unwrap(),
+			StakingLedgerInspect {
+				stash: 11,
+				total: 1000,
+				active: 1000,
+				unlocking: Default::default(),
+				legacy_claimed_rewards: bounded_vec![],
+			}
+		);
+
+		mock::start_active_era(2);
+		assert_eq!(active_era(), 2);
+
+		// Unbond all amount by ensuring chilling
+		let res = Staking::unbond(RuntimeOrigin::signed(11), 1000);
+		assert!(res.is_ok());
+
+		assert_eq!(*staking_events().last().unwrap(), Event::Unbonded { stash: 11, amount: 1000 });
+		assert_eq!(
+			*staking_events().get(staking_events().len() - 2).unwrap(),
+			Event::Chilled { stash: 11 }
+		);
+	})
```

Comment:
nit: would be lot nicer to assert against all events (see [this](https://github.com/paritytech/polkadot-sdk/blob/06e8563b1bac5d490ca3ac058396598141d4a5f2/substrate/frame/staking/src/tests.rs#L3384)) or do a match (see [this](https://github.com/paritytech/polkadot-sdk/blob/06e8563b1bac5d490ca3ac058396598141d4a5f2/substrate/frame/staking/src/tests.rs#L3047))